tools/mo/openvino/tools/mo/moc_frontend/pytorch_frontend_utils.py

```diff
@@ -23,16 +24,68 @@ def remove_tmp_onnx_model(out_dir):
             os.remove(tmp_onnx_model)
 
 
-def get_pytorch_decoder(model, input_shape, example_inputs, input_info):
+
+def get_pytorch_decoder(model, input_shape, example_inputs, args):
     try:
         from openvino.frontend.pytorch.decoder import TorchScriptPythonDecoder
     except Exception as e:
         log.error("PyTorch frontend loading failed")
         raise e
-    inputs = prepare_torch_inputs(example_inputs, input_shape, input_info, allow_none=True)
+    inputs = prepare_torch_inputs(example_inputs, input_shape, args.get("input"), allow_none=True)
     decoder = TorchScriptPythonDecoder(model, example_input=inputs)
+    args['input_model'] = decoder
+    args["framework"] = "pytorch"
+    if inputs is not None:
+        extract_input_info_from_example(inputs, args)
         
-    return decoder
+    return args
+
+
+def extract_input_info_from_example(example_inputs, args):
+    import openvino
+    try:
+        from openvino.frontend.pytorch.decoder import pt_to_ov_type_map
+    except Exception as e:
+        log.error("PyTorch frontend loading failed")
+        raise e
+    inputs = input_to_input_cut_info(args.get("input", []))
+    input_shapes = args.get("input_shape")
+    input_shape_to_input_cut_info(input_shapes, inputs)
+    is_dict_input = isinstance(example_inputs, dict)
+    list_inputs = list(example_inputs.values()) if is_dict_input else example_inputs
```

Comment:
Looks like it partially repeats functionally of this method https://github.com/openvinotoolkit/openvino/blob/80519162ae45021c1df0e516cc94e6aa9cde8ec0/tools/mo/openvino/tools/mo/convert_impl.py#L715

I think we can move extracting inputs info from example_input to this method, to avoid duplicating.